### PR TITLE
Update interfaces for windowed/winning post

### PR DESCRIPTION
**Pull request by magik6k**
Ready for review, but wait for https://github.com/filecoin-project/filecoin-ffi/pull/69 to merge first

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -19,15 +19,11 @@ type Storage interface {
...
-type FallbackPostOut struct {
-	PoStInputs []PoStCandidateWithTicket
-	Proof      []abi.PoStProof
-}
-
 type Prover interface {
-	GenerateEPostCandidates(ctx context.Context, miner abi.ActorID, sectorInfo []abi.SectorInfo, challengeSeed abi.PoStRandomness, faults []abi.SectorNumber) ([]PoStCandidateWithTicket, error)
-	GenerateFallbackPoSt(ctx context.Context, miner abi.ActorID, sectorInfo []abi.SectorInfo, challengeSeed abi.PoStRandomness, faults []abi.SectorNumber) (FallbackPostOut, error)
-	ComputeElectionPoSt(ctx context.Context, miner abi.ActorID, sectorInfo []abi.SectorInfo, challengeSeed abi.PoStRandomness, winners []abi.PoStCandidate) ([]abi.PoStProof, error)
+	GenerateWinningPoStSectorChallenge(proofType abi.RegisteredProof, minerID abi.ActorID, randomness abi.PoStRandomness, eligibleSectorCount uint64) ([]uint64, error)
```
Comment (laser) [inline storage/storage.go]: @magik6k - What does each `uint64` in the returned slice represent? Indices in proving set? sector numbers?

File: storage/storage.go [hunk 2]
```diff
@@ -19,15 +19,11 @@ type Storage interface {
...
+	GenerateWinningPoSt(minerID abi.ActorID, sectorInfo []abi.SectorInfo, randomness abi.PoStRandomness) ([]abi.PoStProof, error)
+
+	GenerateWindowPoSt(minerID abi.ActorID, privateSectorInfo []abi.SectorInfo, randomness abi.PoStRandomness) ([]abi.PoStProof, error)
```
Comment (laser) [inline storage/storage.go]: Nitpick: `GenerateWinningPoSt` and `GenerateWindowPoSt` each have a parameter of type `abi.SectorInfo`. I suggest calling both parameters `sectorInfo`.

**Review by whyrusleeping — APPROVED** [review]
LGTM

File: storage/storage.go (hunk 1 above)
Comment (icorderi) [reply storage/storage.go]: Can we newtype or alias those uint64 going out?

**Review by icorderi — APPROVED** [review]

File: storage/storage.go (hunk 1 above)
Comment (magik6k) [reply storage/storage.go]: This as far I understand is the sector index in the proving partition AMT which doesn't exist in specs-actors yet.

I expect the type to be defined in specs-actors when windowed post is fully implemented in there

**Comment by magik6k** [comment]
(will do the type thing when it's done in specs-actors)

**Pull request merged**